python_modules/dagster/dagster/_core/pipes/utils.py

```diff
@@ -423,14 +468,43 @@ def _logs_thread(
             )
             raise
         finally:
-            for log_reader in self.log_readers:
+            for log_reader in self.log_readers.values():
                 if log_reader.is_running():
                     log_reader.stop()
 
 
+class PipesBlobStoreMessageReader(PipesThreadedMessageReader):
+    """Message reader that reads a sequence of message chunks written by an external process into a
+    blob store such as S3, Azure blob storage, or GCS.
+
+    The reader maintains a counter, starting at 1, that is synchronized with a message writer in
+    some pipes process. The reader starts a thread that periodically attempts to read a chunk
+    indexed by the counter at some location expected to be written by the pipes process. The chunk
+    should be a file with each line corresponding to a JSON-encoded pipes message. When a chunk is
+    successfully read, the messages are processed and the counter is incremented. The
+    :py:class:`PipesBlobStoreMessageWriter` on the other end is expected to similarly increment a
+    counter (starting from 1) on successful write, keeping counters on the read and write end in
+    sync.
+
+    If `log_readers` is passed, the message reader will start the passed log readers when the
+    `opened` message is received from the external process.
+
+    Args:
+        interval (float): interval in seconds between attempts to download a chunk
+        log_readers (Optional[Sequence[PipesLogReader]]): A set of readers for logs.
+    """
+
+    def target_is_readable(self, params: PipesParams) -> bool:
+        return (
+            True  # historical reasons, this method was introduced after the initial implementation
+        )
+
+
 class PipesLogReader(ABC):
     @abstractmethod
-    def start(self, params: PipesParams, is_session_closed: Event) -> None: ...
+    def start(
+        self, params: PipesParams, is_session_closed: Event, state: Optional[Dict[str, Any]] = None
```

Comment:
`state` is worthy of commentary here. It is very generic so difficult to understand its intended use.

Reply:
I think it's actually not needed here. I'll remove it.